recipes/xapian/build.sh

```diff
@@ -0,0 +1,28 @@
+if [[ ${PY3K} == "1" ]]; then
+    PYFLAG="--with-python3"
+    PYTHON2=
+    PYTHON3=$PYTHON
+else
+    PYFLAG="--with-python"
+    PYTHON2=$PYTHON
+    PYTHON3=
+fi
+
+PYTHON2=$PYTHON2 PYTHON3=$PYTHON3 \
+./configure \
+    CPPFLAGS=-I$CONDA_PREFIX/include \
+    LDFLAGS=-L$CONDA_PREFIX/lib \
```

Comment:
There's various conda envs active in the course of a conda build and test. Usually the one you want is `$PREFIX` rather than `$CONDA_PREFIX`. If you need to access language headers or other things installed by `reqs/build`, `$BUILD_PREFIX` is the one to try.

Reply:
Yeah, `zlib.h` is needed during build. I haven't really understood the purpose of `host` vs. `build`, but TBH I feel I shouldn't need to. My situation is pretty standard for a pure C(XX) library: need header and lib during build and lib at runtime.

Reply:
As I understand it, the idea is to separate into `reqs/build` the packages that are consumed by the build process (compilers, make, cmake) from those "ordinary" dependencies in `reqs/host` where (for C(XX) projects) one needs headers at compile-time, libraries at link-time, and libraries at run-time. That view isn't perfect, as compilers aren't wholly consumed, usually conferring a runtime library dependence (e.g., `libgcc-ng`) on the built package, but it's been a workable approximation. The distinction becomes clearer for cross-compiling when native go into `build` and target go into `host`.

In practical terms, except for build tools, put everything in `reqs/host`, and point your `./configure` towards `$PREFIX`. Should work fine for `zlib`. Since [zlib uses run_exports](https://github.com/conda-forge/zlib-feedstock/blob/master/recipe/meta.yaml#L14-L17), by including `zlib` in `reqs/host`, a `reqs/run` pinning to version x.x will be automatically imposed.

Reply:
Thanks so much for the explanation. That clears it up quite a bit (way better than the conda documentation I'd say).